query-service-impl/src/test/java/org/hypertrace/core/query/service/postgres/QueryRequestToPostgresSQLConverterTest.java

```diff
@@ -0,0 +1,1212 @@
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "AND span_name = 'GET /login'' OR tenant_id = ''tenant2'",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithBooleanFilter() {
+    QueryRequest queryRequest =
+        buildSimpleQueryWithFilter(createEqualsFilter("Span.is_entry", true));
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        queryRequest,
+        "Select encode(span_id, 'hex') FROM public.\"span-event-view\" WHERE "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "AND is_entry = 'true'",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithDoubleFilter() {
+    QueryRequest queryRequest =
+        buildSimpleQueryWithFilter(createEqualsFilter("Span.metrics.duration_millis", 1.2));
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        queryRequest,
+        "Select encode(span_id, 'hex') FROM public.\"span-event-view\" WHERE "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "AND duration_millis = 1.2",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithFloatFilter() {
+    QueryRequest queryRequest =
+        buildSimpleQueryWithFilter(createEqualsFilter("Span.metrics.duration_millis", 1.2f));
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        queryRequest,
+        "Select encode(span_id, 'hex') FROM public.\"span-event-view\" WHERE "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "AND duration_millis = 1.2",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithIntFilter() {
+    QueryRequest queryRequest =
+        buildSimpleQueryWithFilter(createEqualsFilter("Span.metrics.duration_millis", 1));
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        queryRequest,
+        "Select encode(span_id, 'hex') FROM public.\"span-event-view\" WHERE "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "AND duration_millis = 1",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithTimestampFilter() {
+    QueryRequest queryRequest =
+        buildSimpleQueryWithFilter(createTimestampFilter("Span.is_entry", Operator.EQ, 123456));
```

Comment:
Shouldn't this use `timestamp` column? Else it is same as `testQueryWithIntFilter`

Reply:
done